pkg/asset/installconfig/azure/metadata.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+
+	typesazure "github.com/openshift/installer/pkg/types/azure"
+)
+
+// Metadata holds additional metadata for InstallConfig resources that
+// does not need to be user-supplied (e.g. because it can be retrieved
+// from external APIs).
+type Metadata struct {
+	session   *Session
+	client    *Client
+	dnsCfg    *DNSConfig
```

Comment:
why create these? users can create these using the session right?

Reply:
Creating these just reduces repeated code. Both `Client` and `DNSConfig` are just wrappers around `Session`, they contain no other state besides a non-exported session. I stopped short of refactoring them away entirely and opted instead to streamline their creation like this.